### ISSUE TITLE
chat.users not work

### ISSUE THREAD

**Issue by tribushevsky** [pubnub/chat-engine-apple#5]
CENChatEngine *client = [CENChatEngine clientWithConfiguration:configuration];
client.connect(userName).perform();
   
client.on(@"$.ready", ^(CENMe *me) {
self.lobby = client.Chat().name(@"some_chat_name").create();

});

// Connect to chat with same name on other devices.        
NSDictionary *users = chat.users; // users contain only current user

What I'm do wrong?

**Comment by parfeon** [comment]
@tribushevsky thanks for reporting about issue.
I was able to track it down and it looks like there is some late cues between PubNub presence service and stream controller which used by ChatEngine.
We are working on solution finding for this.

**Comment by tribushevsky** [comment]
Thanks you for answer!
But how can I get users that was be joined to that chat in past? 
I found - (void)fetchParticipantsForChat:(CENChat *)chat method on CENChatEngine+Private.m file, but after creating a new chat and connect other users to chat with same name as created result.data.uuids contain empty array.

**Comment by parfeon** [comment]
@tribushevsky that is correct API, but it use PubNub presence service API with which I noticed some delays.
Also you can track changes of presence on particular chat by listening [$.online.join](https://github.com/pubnub/chat-engine-apple/wiki/reference-chat#event-online-join), [$.online.here](https://github.com/pubnub/chat-engine-apple/wiki/reference-chat#event-online-here), [$.offline.disconnect](https://github.com/pubnub/chat-engine-apple/wiki/reference-chat#event-offline-disconnect) or [$.offline.leave](https://github.com/pubnub/chat-engine-apple/wiki/reference-chat#event-offline-leave)

But as I mentioned, there is delay in presence service now and event will arrive in about two minutes.
At this moment we trying to find how to fix this delay.

**Comment by nishith-pubnub** [comment]
@madisonsmith we might want to fix this by adding clarity in our docs. Basically chat.users only returns a list of online users NOT all users that are part of a chat group. Customers will need to maintain an aggregated list of users themselves.

**Comment by parfeon** [comment]
@nishith-pubnub developer's issue related to a bit different case (there is ticket for this).

**Comment by nishith-pubnub** [comment]
I'm marking the issue as resolved. Please feel free to contact PubNub Support (https://support.pubnub.com) if you continue to experience issues.
